src/firefly/java/edu/caltech/ipac/astro/ibe/datasource/AtlasIbeDataSource.java

```diff
@@ -0,0 +1,395 @@
+                if (mcen != null && (mcen.equalsIgnoreCase(MCEN) || Boolean.parseBoolean(mcen))) {
+                    queryParam.setMcen(true);
+
+                } else {
+                    // workaround:
+                    if (StringUtils.isEmpty(queryInfo.get("radius"))) {
+                        queryParam.setSize(queryInfo.get("size"));
+                    } else {
+                        queryParam.setSize(queryInfo.get("radius"));
+                    }
+                }
+            }
+        }
+        // process constraints
+        String constraints = processAtlasConstraints(queryInfo);
+        if (!StringUtils.isEmpty(constraints)) {
+            queryParam.setWhere(constraints);
+        }
+
+        return queryParam;
+    }
+
+    private String processAtlasConstraints(Map<String, String> queryInfo) {
+        // create constraint array
+        ArrayList<String> constraints = new ArrayList<String>();
+        String constrStr = "";
+
+        String bands = queryInfo.get(BAND_KEY);
+        // process BAND - ENUMSTRING
+        if (!StringUtils.isEmpty(bands)) {
+            String checkedBands = checkAndConvert2StringSingleQuote(bands);
+            constraints.add("band_name IN (" + checkedBands + ")");
+        }
+        String inst = queryInfo.get(INSTRUMENT_KEY);
+        if (!StringUtils.isEmpty(inst)) {
+            constraints.add("instrument_name=\'" + inst + "\'");
+        }
+
+        // Main image to be displayed is controlled with 'principal' colummn = 1 for each file_type
+        // TODO remove that when 'principal' is available in metadata (IRSA-822) and assign to 1.
+        String pvalue = queryInfo.get(PRINCIPAL_KEY);
+        if (!StringUtils.isEmpty(pvalue)) {
+            constraints.add(PRINCIPAL_KEY + "=" + pvalue);
+        }
+
+        // extra filter is passed from the request client or the example ENUM defined
+        // This can be different for different ATLAS dataset: cosmos, herschel, spitzer, etc.
+        String xtraFilter = ds == null ? queryInfo.get(XTRA_KEY) : ds.getSpecificFilter();
+        if (!StringUtils.isEmpty(xtraFilter)) {
+            constraints.add(xtraFilter);
+        }
+        // compile all constraints
+        if (!constraints.isEmpty()) {
+
+            int i = 0;
+            for (String s : constraints) {
+                if (i > 0) {
+                    constrStr += " AND ";
+                }
+                constrStr += s;
+
+                i++;
+            }
+        }
+        //Overwrite if where key is set:
+        String where = queryInfo.get("where");
+        if (where != null) {
+            constrStr = where;
+        }
+        return constrStr;
+    }
+
+    /**
+     * Check and convert if needed to single quote value to be used in where=band in (...)
+     *
+     * @param str CSV string
+     * @return CSV single quoted values
+     */
+    private String checkAndConvert2StringSingleQuote(String str) {
+        String res = "";
```

Comment:
I am not sure if you did this but I really don't think we should send quoted parameters from the client.  The client should just send a string  and if the irsa server requires it then the search processor should put them on.

Reply:
That's exatcly why i've added this method ;-) The client send regular string (no quotes), but the IBE API needs single quote around values to express the where clause.

Reply:
oh, good!